examples/tv-casting-app/linux/main.cpp

```diff
@@ -218,6 +241,32 @@ void InitCommissioningFlow(intptr_t commandArg)
     }
 }
 
+CHIP_ERROR DiscoverCommissioners()
+{
+    // Send discover commissioners request
+    ReturnErrorOnFailure(gCommissionableNodeController.DiscoverCommissioners(gDiscoveryFilter));
+
+    // Give commissioners some time to respond and then ScheduleWork to initiate commissioning
+    DeviceLayer::SystemLayer().StartTimer(
+        chip::System::Clock::Milliseconds32(kCommissionerDiscoveryTimeoutInMs),
+        [](System::Layer *, void *) { chip::DeviceLayer::PlatformMgr().ScheduleWork(InitCommissioningFlow); }, nullptr);
+
+    return CHIP_NO_ERROR;
+}
+
+CHIP_ERROR RequestCommissioning(int index)
+{
+    const Dnssd::DiscoveredNodeData * selectedCommissioner =
+        gCommissionableNodeController.GetDiscoveredCommissioner(index);
+    if (selectedCommissioner == nullptr)
+    {
+        ChipLogError(AppServer, "No such commissioner!");
```

Comment:
I would add details if we log: `ChipLogError(AppServer, "No commisionier with index %d exists", index);`

Reply:
Done